src/openrct2-unity/Assets/Scripts/Tools/SelectionTool.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using OpenRCT2.Unity;
+
+public class SelectionTool : MonoBehaviour
+{
+
+    [SerializeField] private CanvasManager canvasManager;
+    [SerializeField] private PeepController peepController;
+
+
+    void Start()
+    {
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        if (Input.GetMouseButtonDown(0))
+        {
+            var ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+            RaycastHit hit;
+            if (Physics.Raycast(ray, out hit))
+            {
+                CheckHit(hit.collider.gameObject);
+            }
+        }
+    }
+
+    void CheckHit(GameObject obj)
+    {
+        switch (obj.name)
+        {
+            case string a when a.Contains("Guest"):
```

Comment:
Might be better to use Unity tags for this: https://docs.unity3d.com/Manual/Tags.html instead of the name. This also allows for a literal compare instead of using 'Contains'. Also the string pattern matching is not necessary because both name and tag are already a string.

Reply:
True, will change it.